custom_components/givenergy_local/__init__.py

```
@@ -25,16 +92,173 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+                ]
+                if not coordinators:
+                    raise HomeAssistantError("No connected GivEnergy inverter found")
+
+            for coordinator in coordinators:
+                if coordinator.data is None:
+                    continue
+                inv = coordinator.data.inverter.serial_number.lower()
+                frames: list[str] = []
+
+                def _sink(direction: str, data: bytes) -> None:
+                    frames.append(f"{direction}: {data.hex()}")
```

Ruff (0.15.13)

[warning] Function definition does not bind loop variable `frames`

(B023)